File: 1-50/4/main.rb
#!/usr/bin/env ruby

900.upto(999).each do |a|
  900.upto(999).each do |b|
    0.upto(9).each do |e|
      0.upto(9).each do |f|
        n = 900009 + 10010 * e + 1100 * f
        if n == a * b
          p "#{n} = #{a} * #{b}"
        end
      end
    end
  end
end

File: 1-50/4/main.py
#!/usr/bin/env python

# 9ab * 9cd = 9effe9

for a in range(900, 1000):
    for b in range(900, 1000):
        for e in range(0, 10):
            for f in range(0, 10):
                num = 900009 + 10010*e + 1100*f
                if num == a * b:
                    print 'found!  %d = %d * %d' % (num, a, b)
